src/language/statics/Statics.re

```diff
@@ -1424,10 +1456,12 @@ and upat_to_info_map =
           id: Pat.rep_id(upat),
           typ: ctx_typ,
         });
+      let warning: Warning.t = Warning.var_is_unused(co_ctx, name);
```

Comment:
to calculate this warning it looks like you need the co_ctx and the name; both of are already captured by the info; you can add this line to Info.derived_pat (you'll need a switch there to to get the name from Var(name)).

I'm think if you do it this way you might not need to thread `~warning` around 20 times in this file.